.github/workflows/python-package.yml

```diff
@@ -17,11 +17,16 @@ on:
   merge_group:
     types: [checks_requested]
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}-${{ github.head_ref }}
+  cancel-in-progress: ${{ ! (github.ref == 'refs/heads/master' || startsWith(github.ref, 'refs/heads/release/')) }}
```

Comment:
I don't know the meaning of `github.ref` and whether 'refs/heads/master' is a valid value. Our main branch is "main" in case they are related.